### DISCUSSION TITLE
Unwrap twice

### DISCUSSION THREAD

**Discussion by van-sprundel** [tokio-rs/toasty#41] [Q&A] [answered]
Hey, love the concept!

Had a question about the following snippet:
```rust
while let Some(todo) = todos.next().await {
    let todo = todo.unwrap();
    ...
}
```

I was curious why the design makes it necessary to unwrap twice. Can an item exist but be invalid?

**Comment by carllerche** [comment, marked as answer]
There is only one unwrap there? I think it is because `next()` returns `Option<Result<_>>`. The `Option` returning `None` indicates the stream is empty. `Result` tracks if there is an error or not.
